python/mxnet/image/image.py

```diff
@@ -1149,22 +1158,44 @@ def __init__(self, batch_size, data_shape, label_width=1,
         else:
             self.auglist = aug_list
         self.cur = 0
+        self._is_allowed_reading = True
+        self._cached_data = None
+        # handle the last batch
+        if self.seq and last_batch == 'discard':
+            new_seq_n = len(self.seq) - len(self.seq) % batch_size
+            self.seq = self.seq[:new_seq_n]
```

Comment:
this is not correct, you now change the behavior from initial shuffle to per epoch shuffle, so self.seq must be shuffled in reset, and the slice can not be fixed.

Reply:
According to (NDArrayIter)[https://mxnet.incubator.apache.org/_modules/mxnet/io.html#NDArrayIter], it shuffled only during the initialization. But you are right. We should shuffle for each epoch.